app/lib/actions/katello/repository/metadata_generate.rb

```diff
@@ -15,14 +15,16 @@ def plan(repository, options = {})
           else
             matching_content = options.fetch(:matching_content, false)
           end
+          publication_content_type = !(repository.content_type == "ansible_collection")
 
           plan_pulp_action([Pulp::Repository::DistributorPublish, Pulp3::Orchestration::Repository::GenerateMetadata],
                         repository, SmartProxy.pulp_master,
                         :force => force,
                         :source_repository => source_repository,
                         :matching_content => matching_content,
                         :dependency => dependency,
-                        :repository_creation => repository_creation)
+                        :repository_creation => repository_creation,
+                        :publication_content_type => publication_content_type)
```

Comment:
any reason this needs to be determined here?  can it just be handled in the pulp3 orchestration layer?